### ISSUE TITLE
CI fails with "Could not find type GdUnitHtmlReport"

### ISSUE THREAD

**Issue by jeremyhcho** [MikeSchulze/gdUnit4#345]
### The used GdUnit4 version

4.2.1 (Pre Release/Master branch) + 4.2.0

### The used Godot version

4.2.1.stable

### Operating System

Both `ubuntu-latest` and `ubuntu-22.04`

### Describe the bug

I've added a workflow to run unit tests taking from the sample yml file provided. When I run it though, it errors with
```
Godot Engine v4.2.1.stable.official.b09f793f5 - https://godotengine.org/
WARNING: Could not set V-Sync mode. V-Sync is not supported.
     at: set_use_vsync (platform/linuxbsd/x[11](https://github.com/tbd-interactive/eclipse/actions/runs/7879537212/job/21499983839?pr=66#step:5:12)/gl_manager_x11.cpp:374)
OpenGL API 4.5 (Core Profile) Mesa 23.0.4-0ubuntu1~22.04.1 - Compatibility - Using Device: Mesa - llvmpipe (LLVM 15.0.7, 256 bits)
 
Debugger Break, Reason: 'Parser Error: Could not find type "GdUnitHtmlReport" in the current scope.'
*Frame 0 - res://./addons/gdUnit4/bin/GdUnitCmdTool.gd:27 in function ''
```

The Github action yml file
```
name: Run Unit Tests
run-name: ${{ github.head_ref || github.ref_name }}-test

on:
  pull_request:
    paths-ignore:
      - '**.yml'
      - '**.md'
  workflow_dispatch:


concurrency:
  group: test${{ github.event.number }}
  cancel-in-progress: true


jobs:
  unit-test:
    strategy:
      fail-fast: false
      matrix:
        godot-version: ['4.2.1'] # Insert here the Godot version you want to run your tests with

    name: "CI Unit Test 🐧 v${{ matrix.godot-version }}"
    runs-on: ubuntu-22.04
    timeout-minutes: 10 # The overall timeout

    steps:
      - name: "Checkout"
        uses: actions/checkout@v3
        with:
          lfs: true

      - name: "🐧Download and Install Godot ${{ matrix.godot-version }}"
        continue-on-error: false
        shell: bash
        run: |
          GODOT_HOME=$HOME/bin
          GODOT_BIN=$GODOT_HOME/godot
          mkdir -p $GODOT_HOME
          chmod 770 $GODOT_HOME
          GODOT_CONF_PATH=$HOME/.config/godot
          if [ ! -d $GODOT_CONF_PATH ]; then
            mkdir -p $GODOT_CONF_PATH
            chmod 770 $GODOT_CONF_PATH
          fi

          GODOT_PACKAGE=Godot_v${{ matrix.godot-version }}-stable_linux.x86_64
          wget https://github.com/godotengine/godot/releases/download/${{ matrix.godot-version }}-stable/$GODOT_PACKAGE.zip -P ${{ runner.temp }}
          unzip ${{ runner.temp }}/$GODOT_PACKAGE.zip -d $GODOT_HOME
          mv $GODOT_HOME/$GODOT_PACKAGE $GODOT_BIN
          chmod u+x $GODOT_BIN
          echo "GODOT_HOME=$GODOT_HOME" >> "$GITHUB_ENV"
          echo "GODOT_BIN=$GODOT_BIN" >> "$GITHUB_ENV"

      # We need to update the project before running tests, Godot has actually issues with loading the plugin
      - name: "Update Project"
        if: ${{ !cancelled() }}
        timeout-minutes: 1
        continue-on-error: true # we still ignore the timeout, the script is not quit and we run into a timeout
        shell: bash
        run: |
          ${{ env.GODOT_BIN }} -e --path . -s res://addons/gdUnit4/bin/ProjectScanner.gd --headless --audio-driver Dummy

      - name: "Run Unit Tests"
        if: ${{ !cancelled() }}
        timeout-minutes: 8 # set your expected test timeout
        env:
          GODOT_BIN: ${{ env.GODOT_BIN }}
        shell: bash
        run: |
          chmod +x addons/gdUnit4/runtest.sh
          xvfb-run --auto-servernum addons/gdUnit4/runtest.sh --add "res://tests" --audio-driver Dummy --display-driver x11 --rendering-driver opengl3 --screen 0 --continue

      - name: "Publish Test Report"
        if: ${{ always() }}
        uses: dorny/test-reporter@v1.6.0
        with:
          name: "test_report_${{ matrix.godot-version }}"
          path: "reports/**/results.xml"
          reporter: java-junit
          fail-on-error: 'false'

      - name: "Upload Unit Test Reports"
        if: ${{ always() }}
        uses: actions/upload-artifact@v3
        with:
          name: "test_report_${{ matrix.godot-version }}"
          path: |
            reports/**
            /var/lib/systemd/coredump/**
            /var/log/syslog

  finalize:
    if: ${{ !cancelled() }}
    runs-on: ubuntu-latest
    name: Final Results
    needs: [unit-test]
    steps:
      - run: exit 1
        if: >-
          ${{
              contains(needs.*.result, 'failure')
            || contains(needs.*.result, 'cancelled')
          }}
```

I've tried with both gdunit 4.2.0 and 4.2.1

**Comment by MikeSchulze** [comment]
Hi, it should work, on which step the error occurs? 

The documentation is not updated yet, but I provided a public [gdunit-action](https://github.com/MikeSchulze/gdUnit4-action) for easier CI setup.

I would suggest using the new action instead of building manually the ci workflow.
Your steps should look like 
```
    steps:
      - name: 'Checkout'
        uses: actions/checkout@v4
        with:
          lfs: true            

      - name: 'Test GdUnit4 ${{ matrix.version }} - Godot_${{ matrix.godot-version }}-${{ matrix.godot-status }}'
        uses: MikeSchulze/gdUnit4-action@v1.0.1
        with:
          godot-version: ${{ matrix.godot-version }}
          godot-status: ${{ matrix.godot-status }}
          version: ${{ github.head_ref }}
          paths: |
            res://tests/
          timeout: 10
          report-name: report_${{ matrix.version }}_Godot${{ matrix.godot-version }}-${{ matrix.godot-status }}.xml
```

**Comment by jeremyhcho** [comment]
Ah neat, didn't know there was an action. Thanks!

I did just try using it though and now I'm running into this error
```
Delete 'addons/gdUnit4/test'
 Install GdUnit4 master plugin in the project 
mv: cannot move './.install-gdunit4/addons' to './addons': Directory not empty
```

Seems like from this line https://github.com/MikeSchulze/gdUnit4-action/blob/master/action.yml#L112 but I'm not sure what's causing it

**Comment by MikeSchulze** [comment]
Arg, looks like this is a bug in the action. I guess your `addons` folder has installed additional plugins.
What does your project structure look like?

**Comment by MikeSchulze** [comment]
Is fixed use `MikeSchulze/gdUnit4-action@v1.0.2`
